### PR TITLE
[HYDRATOR-1022] [HYDRATOR-1024] Made file tracking table and its expiry time input fields optional in XMLReader plugin

### PR DESCRIPTION
**Pull request by romy-khetan**

### CODE REVIEW

**Comment by russorat** [comment]
hydrator_studio
i still see the table expiry days as required. did you test this?

File: core-plugins/widgets/XMLReader-batchsource.json [hunk 1]
```diff
@@ -67,7 +67,6 @@
           "label": "Table Data Expiry Period (Days)",
           "name": "tableExpiryPeriod",
           "widget-attributes": {
-            "default": "30",
             "min": 1,
```
Comment (russorat) [inline core-plugins/widgets/XMLReader-batchsource.json]: having min and max here is causing this to show up as required in the UI. remove these and verify the value is null or > 0 in the validate method.

File: core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/XMLReaderBatchSource.java [hunk 2]
```diff
@@ -253,11 +259,14 @@ public void onRunFinish(boolean succeeded, BatchSourceContext context) {
...
     @Macro
     private final String tableName;
 
-    @Description("Expiry period (days) for data in the table. Default is 30 days. " +
+    @Nullable
+    @Description("Expiry period (days) for data in the table. Data will be persisted in the table " +
+      "if no expiry period is provided." +
       "Example: For tableExpiryPeriod = 30, data before 30 days get deleted from the table.")
     @Macro
     private final String tableExpiryPeriod;
```
Comment (russorat) [inline core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/XMLReaderBatchSource.java]: This should be an Integer

File: core-plugins/widgets/XMLReader-batchsource.json [hunk 3]
```diff
@@ -67,7 +67,6 @@
           "label": "Table Data Expiry Period (Days)",
```
Comment (russorat) [inline core-plugins/widgets/XMLReader-batchsource.json:67]: the widget type should be a textbox. there is a bug with the number widet-type: https://issues.cask.co/browse/HYDRATOR-1187

File: core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/XMLReaderBatchSource.java [hunk 4]
```diff
@@ -310,12 +319,6 @@ void validate() {
...
       }
-      if (!containsMacro("tableName")) {
-        Preconditions.checkArgument(!Strings.isNullOrEmpty(tableName), "Table name cannot be empty.");
-      }
-      if (!containsMacro("tableExpiryPeriod")) {
-        Preconditions.checkArgument(tableExpiryPeriod != null, "Table expiry period cannot be empty.");
-      }
       if (!containsMacro("temporaryFolder")) {
         Preconditions.checkArgument(!Strings.isNullOrEmpty(temporaryFolder), "Temporary folder cannot be empty.");
       }
```
Comment (russorat) [inline core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/XMLReaderBatchSource.java:329]: add a check to verify the tableExpiryPeriod is > 0 if it is not null and does not contain a macro.

**Comment by romy-khetan** [comment]
@russorat Updated the PR. Please review.

**Comment by russorat** [comment]
LGTM

**Comment by abhinavbansal246** [comment]
@romy-khetan Unit test for xml reader is failing
https://travis-ci.org/caskdata/hydrator-plugins/builds/188222497#L759

**Comment by abhinavbansal246** [comment]
@romy-khetan Did we merge develop into this branch because we had a rebase already?
We prefer rebase over merge

**Comment by romy-khetan** [comment]
@abhinavbansal246 
Updated PR and used rebase instead of merge. Please review

**Pull request merged**